### PR TITLE
fix detection of thin staves (probably) in vector graphs

### PR DESCRIPTION
**Pull request by liang-chen**

### CODE REVIEW

File: mscore/musescore.cpp [hunk 1]
```diff
@@ -4777,8 +4777,8 @@ void MuseScore::cmd(QAction* a, const QString& cmd)
             if (_textTools)
                   _textTools->toggleUnderline();
             }
-      else if (cmd == "edit-toolbars")
-            showToolbarEditor();
+      //else if (cmd == "edit-toolbars")
+            //showToolbarEditor();
```
Comment (Jojo-Schmitz) [inline mscore/musescore.cpp]: Why?

Comment (liang-chen) [inline mscore/musescore.cpp]: This was for debugging and has already been recovered.

**Comment by lasconic** [comment]
Can you explain the rationale of these changes? You removed the border detection and then tweaked the  staff detection. Are both needed to improve the staff detection?

**Comment by liang-chen** [comment]
The border removal is not needed any more since we have clean vector graph to binary image conversion right now. Previously, after image conversion there's always some black border surrounding the converted image, so I wrote this function to simply remove the borders.

For staff detection, the searching range was too wide so when we have thin staves it tends to detect a unreasonably large spatium, so I restrict this range and it seems to work. A better solution is to multiply the autocorrelation with a weighting function which takes the prior confidence of spatium value instead of cutting of the max spatium in an ad hoc way. In addition, could you tell me what the function xproject2() does? I haven't quite figured it out yet.

**Comment by lasconic** [comment]
From Werner
xproject2 tries to find a horizontal line by doing a projection of the x-axis counting on/off bits

**Comment by liang-chen** [comment]
Okay this makes sense to me, but the previous range was still unreasonably large -- it searches up to 100 pixels for the spatium; now the range is btw 5 and 50, and solves some problems as far as I've observed.

**Pull request merged**